src/main/java/edu/texas/social_computing/hospitals/Matching.java

```diff
@@ -44,4 +43,46 @@ public boolean isFull(Hospital h) {
         return getAssignedResidents(h).size() == h.getCapacity();
     }
 
+    public Set<Resident> getAllUnassigned(List<Resident> residents) {
+        Set<Resident> unassignedResidents = new HashSet<>();
+        for(Resident resident : residents) {
+            if(!hasAssignment(resident)) {
+                unassignedResidents.add(resident);
+            }
+        }
+        return unassignedResidents;
+    }
+
+    /**
+     *
+     * @param residents : a list of Non-Dominant (ND) residents in a matching that needs to be checked to see
+     *                 if any of the residents in the list violate a proximity constraint
+     * @return a set of residents that violate the proximity constraint AND are the worse placed partner
+     */
+    public Set<Resident> getNDProximityViolators(List<Resident> residents, ResidentTable residentTable) {
+        Set<Resident> freeViolatingResidents = new HashSet<>();
+        for(Resident resident : residents) {
+            if(!resident.hasPartner()) continue;
+            Resident partner = residentTable.getResidentById(resident.getPartnerId());
+            Hospital assignedHospital = this.getAssignedHospital(resident);
+            Hospital partnerAssignedHospital = this.getAssignedHospital(partner);
+            if(assignedHospital == Matching.NO_MATCH) continue;
+            if(partnerAssignedHospital == Matching.NO_MATCH) {
+                freeViolatingResidents.add(partner);
+                continue;
+            }
```

Comment:
can we change this to look like:
```
if(!hasAssignment(resident)) continue;
if(!hasAssignment(partner))  {
    freeViolatingResidents.add(partner);
    continue;
}
Hospital assignedHospital = this.getAssignedHospital(resident);
Hospital partnerAssignedHospital = this.getAssignedHospital(partner);
```

I know it is very similar, but then just a little more clear about what the `if` is checking.